abel/tools/circularize.py

```diff
@@ -0,0 +1,236 @@
+
+    Parameters
+    ----------
+    IM : numpy 2D array
+
+    method: str
+        method to align slice profiles,
+        "argmax" - compare intensity-profile.argmax() of each radial slice.
+        "lsq" - determine radial scaling factor by minimizing the
+                intensity-prfoile with an adjacent slice intensity-profile.
+
+    center: str, or None
+        pre-center image using PyAbel centering methods,
+        "com", "convolution", "gaussian", "image_center", "slice"
+
+    radial_range: tuple, or None
+        limit slice comparison to the radial range (rmin, rmax).
+
+    zoom: float
+        nmimage.zoom image before analysis. This may help improve the low
+        radius slice profile.
+
+    smooth: float
+        smoothing for spline interpolation of the determined radial scaling
+        factor vs angle
+
+    nslices: int
+        divide the image into nslices. Ideally, this should be a divisor of
+        the image column width.
+
+    ref_angle: None or float
+        reference angle for which radial coordinate is unchanged.
+        None uses numpy.mean(radial scale factors)
+
+    inverse: bool
+        inverse Abel transform the *polar* image, to remove the background
+        intensity. This helps follow a given peak intensityr, for all angles,
+        when the anisotropy parameter is large.
+
+    return_correction: bool
+        additional outputs, see below
+
+    Returns
+    -------
+    IMcirc : numpy 2D array, same size as input
+        Circularized imput image
+
+    (if return_correction is True)
+    slice_angles: numpy 1D array
+        Mid-point angle (radians) of image slice
+
+    radial_corrections: numpy 1D array
+        radial correction scale factors, at each angle
+
+    spline_function: numpy func()
+        spline function used to evaluate the radial correction at each angle
+
+    """
+
+    if zoom > 1:
+        # zoom may improve the low radii angular resolution
+        IM = ndimage.zoom(IM, zoom)
+
+    if center is not None:
+        # convenience function for the case image is not centered
+        IM = abel.tools.center.center_image(IM, center=center)
+
+    # map image into polar coordinates - much easier to slice
+    # cartesian (Y, X) -> polar (Radius, Theta)
+    polarIM, radial_coord, angle_coord =\
+                           abel.tools.polar.reproject_image_into_polar(IM)
+ 
+    if inverse:
+        # pseudo inverse Abel transform of the polar image, removes background
+        # to enhance transition peaks
+        polarIM = abel.dasch.two_point_transform(polarIM.T).T
+
+    # limit radial range of polar image, if selected
+    radial = radial_coord[:, 0]
+    if radial_range is not None:
+        subr = np.logical_and(radial > radial_range[0]*zoom,
+                              radial < radial_range[1]*zoom)
+        polarIM = polarIM[subr]
+        radial = radial[subr]
+
+    # split image into n-slices
+    slices = np.array_split(polarIM, nslices, axis=1)
```

Comment:
Would it be more straightforward to just utilize the `dt` keyword of `reproject_image_into_polar` to generate an image with the appropriate number of slices?  It would probably shorten the code here quite a bit.